core/modules/menu/js/menu.admin.js

```diff
@@ -19,6 +19,62 @@ Backdrop.behaviors.menuAdminFieldsetSummaries = {
   }
 };
 
+Backdrop.behaviors.menuEditItemParents = {
+  attach: function (context, settings) {
+    $('.form-item-parent-menu', context).once('form-item-parent-menu', function () {
+      // Get the menu parent options from settings.
+      var menuOptions = settings.menu_edit_item_parents;
+      // Get the current menu name from settings.
+      var menuName = settings.menu_edit_item_menu;
+      // Always move the menu selector before the menu parent selector.
+      $('.form-item-parent-menu').insertBefore('#menu-parent-select-wrapper');
+      // On load set the current menu as default in the menu select.
+      Backdrop.menu_edit_update_parent_list(menuName);
+      // Ensure that the menu parent select list is filtered again when any ajax
+      // process runs.
+      $(document).ajaxComplete(function(event, xhr, settings) {
+        if (settings.url == Backdrop.settings.basePath + 'system/ajax') {
+          var selected = $('[data-menu-parent] :selected').val().split(':')[0];
+          Backdrop.menu_edit_update_parent_list(selected);
+        }
+      });
+      // On changing the menu select, update the menu parent select.
+      var sel = $('.form-item-parent-menu select');
+      sel.on('change', function () {
+        Backdrop.menu_edit_update_parent_list(this.value);
+      });
+    });
+  }
+}
+
+/**
+ * Function to set the options of the menu parent item dropdown.
+ */
+Backdrop.menu_edit_update_parent_list = function (value) {
+  var values = [value];
+
+  var url = Backdrop.settings.basePath + 'admin/structure/menu/parents';
+  $.ajax({
+    url: location.protocol + '//' + location.host + url,
+    type: 'POST',
+    data: {'menus[]' : values},
+    dataType: 'json',
+    success: function (options) {
+      // Save key of last selected element.
+      var selected = $('[data-menu-parent] :selected').val();
+      // Remove all existing options from dropdown.
+      var selectForm = $('[data-menu-parent]');
+      selectForm.children().remove();
+      // Add new options to dropdown.
+      $.each(options, function(index, value) {
```

Comment:
This looks like odd (uncommon) usage of jQuery `.each()` - see https://api.jquery.com/each/

I'm pretty sure, this could be done with a JS native method. Like `for... in` or Arrray.forEach().

Reply:
This is somewhat common. I counted about a dozen other instances like this in core. There's another usage of this down below only it's `jQuery.each()`. I don't think it's a show-stopper for this PR.